shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/Spark311Shims.scala

```diff
@@ -275,12 +275,19 @@ class Spark311Shims extends Spark301Shims {
         }),
       GpuOverrides.exec[InMemoryTableScanExec](
         "Implementation of InMemoryTableScanExec to use GPU accelerated Caching",
-        ExecChecks(TypeSig.commonCudfTypes + TypeSig.DECIMAL, TypeSig.all),
+        ExecChecks((TypeSig.commonCudfTypes + TypeSig.DECIMAL.withPsNote(TypeEnum.DECIMAL,
+          "Negative scales aren't supported at the moment even with " +
+            "spark.sql.legacy.allowNegativeScaleOfDecimal set to true. This is because Parquet " +
+            "doesn't support negative scale for decimal values") + TypeSig.STRUCT).nested(),
```

Comment:
I think it would be easier to see the supported types if the PS note wasn't smack in the middle of the list, e.g.:
```suggestion
        ExecChecks((TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT).nested()
            .withPsNote(TypeEnum.DECIMAL,
                "Negative scales aren't supported at the moment even with " +
                "spark.sql.legacy.allowNegativeScaleOfDecimal set to true. Parquet " +
                "doesn't support negative scale for decimal values"),
```